src/IPToken.sol

```diff
@@ -13,75 +14,67 @@ struct Metadata {
 }
 
 error TokenCapped();
-error OnlyIssuerOrOwner();
 
 /**
- * @title IPToken
- * @author molecule.to
- * @notice this is a template contract that's spawned by the Tokenizer
- * @notice the owner of this contract is always the Tokenizer contract.
- *         the issuer of a token bears the right to increase the supply as long as the token is not capped.
+ * @title IPToken 1.3
+ * @author molecule.xyz
+ * @notice this is a template contract that's cloned by the Tokenizer
+ * @notice the owner of this contract is always the Tokenizer contract which enforces IPNFT holdership rules.
+ *         The owner can increase the token supply as long as it's not explicitly capped.
  * @dev formerly known as "molecules"
  */
 contract IPToken is ERC20BurnableUpgradeable, OwnableUpgradeable {
     event Capped(uint256 atSupply);
 
-    //this will only go up.
+    /// @notice the amount of tokens that ever have been issued (not necessarily == supply)
     uint256 public totalIssued;
-    /**
-     * @notice when true, no one can ever mint tokens again.
-     */
+
+    /// @notice when true, no one can ever mint tokens again.
     bool public capped;
+
     Metadata internal _metadata;
 
-    function initialize(string calldata name, string calldata symbol, Metadata calldata metadata_) external initializer {
+    function initialize(uint256 ipnftId, string calldata name, string calldata symbol, address originalOwner, string memory agreementCid)
+        external
+        initializer
+    {
         __Ownable_init();
         __ERC20_init(name, symbol);
-        _metadata = metadata_;
+        _metadata = Metadata({ ipnftId: ipnftId, originalOwner: originalOwner, agreementCid: agreementCid });
```

Comment:
shouldn't we have the originalIpmftOwner just like we have the ipnftId?

Reply:
?

Reply:
I meant the naming: we name originalIpnftOwner instead of originalOwner